service/index/reader.go

```diff
@@ -63,6 +63,13 @@ func (r *Reader) Last() (uint64, error) {
 	return height, err
 }
 
+// LatestRegisterHeight returns the latest height for which all registers are indexed
+func (r *Reader) LatestRegisterHeight() (uint64, error) {
```

Comment:
Maybe address in a separate PR, I think the latest register height should be saved in pebble along with the register values, rather than in badger. Otherwise, it might cause data inconsistency